homeassistant/components/devolo_home_control/__init__.py

```diff
@@ -0,0 +1,94 @@
+"""The devolo_home_control integration."""
+from devolo_home_control_api.homecontrol import HomeControl
+from devolo_home_control_api.mydevolo import (
+    Mydevolo,
+    WrongCredentialsError,
+    WrongUrlError,
+)
+import voluptuous as vol
+
+from homeassistant.components import switch as ha_switch
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, EVENT_HOMEASSISTANT_STOP
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .config_flow import create_config_flow
+from .const import DEFAULT_MPRM, DEFAULT_MYDEVOLO, DOMAIN, PLATFORMS
+
+SUPPORTED_PLATFORMS = [ha_switch.DOMAIN]
+
+SERVER_CONFIG_SCHEMA = vol.Schema(
+    vol.All(
+        {
+            vol.Optional("mydevolo", default=DEFAULT_MYDEVOLO): cv.string,
+            vol.Optional("mprm", default=DEFAULT_MPRM): cv.string,
+            vol.Required(CONF_PASSWORD): cv.string,
+            vol.Required(CONF_USERNAME): cv.string,
+        }
+    )
+)
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: SERVER_CONFIG_SCHEMA}, extra=vol.ALLOW_EXTRA)
```

Comment:
Is it possible to hide some options behind an 'extended' button or something similar? We decided to go this way, because the both options which are configured in the yaml file are just for beta testers. So the main user will never need to touch the yaml file, to configure our integration

Reply:
Optional options can go into a options flow:
https://developers.home-assistant.io/docs/config_entries_options_flow_handler

Reply:
Or rather, optional options *should* go into an options flow.

Reply:
The word optional doesn't match exactly. For 99,9% percent of user we can use the default value, but the rest need to adjust two urls to get to the correct stage. So for the 0,1% it is *not* an optional option, because we need to know the URL before login, to know which URL/stage to use.